File: src/models/identity.ts
import { IdentityDto } from "./identity-dto";

export class Identity {
  readonly userId: string;
  readonly account: string;
  readonly arn: string;

  constructor(dto: IdentityDto) {
    this.userId = dto.user_id;
    this.account = dto.account;
    this.arn = dto.arn;
  }
}

File: src-tauri/src/models/identity.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct Identity {
    pub user_id: String,
    pub account: String,
    pub arn: String,
}

impl Identity {
    pub fn new(user_id: &str, account: &str, arn: &str) -> Self {
        Self {
            user_id: user_id.to_string(),
            account: account.to_string(),
            arn: arn.to_string(),
        }
    }
}
